### PR TITLE
Add to the Downward API the support of status.hostIP field

### PR DESCRIPTION
**Pull request by JulienBalestra**
**What this PR does / why we need it**:

This PR allows to use the `status.hostIP` field in the downward API area.

Global host networks services are running over the k8s daemonSets like Heka for the logs.
The Heka is able to received over UDP / TCP logs from Pods.
The easy way to reach the host is by the downward API.
Currently, the downward API only supports the `spec.nodeName` that will obviously cost DNS operations.

**Special notes for your reviewer**:

This is an example of the usage area of my PR: https://github.com/JulienBalestra/enjoliver/blob/ebc091f7c424d3f51c08dede6d88d16bf2babf60/app/tests/euid/manifests/httpd-daemonset.yaml#L35-L38

I select `hostIP` instead of `nodeIP` to match the current API:

```
curl -s 127.0.0.1:8080/api/v1/pods | jq .items[0].status.hostIP
"172.20.0.89"
```

### CODE REVIEW

File: pkg/kubelet/kubelet_pods.go [hunk 1]
```diff
@@ -619,6 +619,8 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod
 		return pod.Spec.ServiceAccountName, nil
 	case "status.podIP":
 		return podIP, nil
+	case "status.hostIP":
+		return kl.nodeIP.String(), nil
```
Comment (xiangpengzhao) [inline pkg/kubelet/kubelet_pods.go]: `kl.nodeIP` would be nil if you don't set `--node-ip` flag for kubelet. In this case, kubelet will [setNodeAddress](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L440-L443) by other ways. Can we get `hostIP` from `kl.nodeIP` then?

Comment (xiangpengzhao) [inline pkg/kubelet/kubelet_pods.go]: I suggest to use [GetHostIP](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_getters.go#L218) instead.

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

**Comment by xiangpengzhao** [comment]
Duplication of https://github.com/kubernetes/kubernetes/pull/42717 ...

**Comment by JulienBalestra** [comment]
Thank you @xiangpengzhao

I git cherry-pick the @andrewsykim part of works from #42717 

I'm sure we doesn't need to change anything in the [rkt](https://github.com/kubernetes/kubernetes/pull/42717/files#diff-717f367da08d0bb8eb8c45d38f164eb8) parts. I already have this patch in production and it works.

What do you think about it @andrewsykim ?

**Comment by andrewsykim** [comment]
@JulienBalestra most of the work is already done, just waiting on some people to review it. I think its safe to close this PR.

**Pull request closed without merging**